Mathlib/RingTheory/Polynomial/DegreeLT.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Kenny Lau
+-/
+
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.LinearAlgebra.Matrix.Block
+import Mathlib.Algebra.Polynomial.Div
+import Mathlib.Algebra.Polynomial.Taylor
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Polynomials with degree less than `n`
+
+This file contains the properties of the submodule of polynomials of degree less than `n` in a
+commutative ring `R`, denoted `R[X]_n`.
+
+## Main definitions/lemmas
+
+* `degreeLT.basis R n`: a basis for `R[X]_n` the submodule of polynomials with degree `< n`,
+given by the monomials `X^i` for `i < n`.
+
+* `degreeLT.basis_prod R m n`: a basis for `(R[X]_m) × (R[X]_n)` as above, which is the sum
+of the two bases.
+
+* `degreeLT.addLinearEquiv R m n`: an isomorphism between `(R[X]_(m+n))` and `(R[X]_m) × (R[X]_n)`
+given by the fact that the bases are both indexed by `Fin (m+n)`.
+
+* `taylorLinear r n`: The linear automorphism induced by `taylor r` on `R[X]_n`.
+
+-/
+
+
+namespace Polynomial
+
+/-- R[X]_n is notation for the submodule of polynomials of degree strictly less than n. -/
+scoped notation:9000 R "[X]_" n => Polynomial.degreeLT R n
+
+namespace degreeLT
+
+variable {R : Type*} [Semiring R] {m n : ℕ} (i : Fin n) (P : R[X]_n)
+
+variable (R) in
+/-- Basis for R[X]_n given by the `X^i` with `i < n`. -/
+noncomputable def basis (n : ℕ) : Basis (Fin n) R R[X]_n :=
+  .ofEquivFun (Polynomial.degreeLTEquiv R n)
+
+instance : Module.Finite R R[X]_n :=
+  .of_basis <| basis ..
+
+instance : Module.Free R R[X]_n :=
+  .of_basis <| basis ..
+
+lemma X_pow_mem_degreeLT (i : Fin n) : X ^ i.val ∈ R[X]_n := by
+  nontriviality R using mem_degreeLT, WithBot.bot_lt_iff_ne_bot
+  simp only [Polynomial.mem_degreeLT, Polynomial.degree_X_pow, Nat.cast_lt, Fin.is_lt]
+
+variable (R) in
+/-- The element `X^i` in `R[X]_n`. This is equal to `basis R n i`. -/
+noncomputable def xPow (n : ℕ) (i : Fin n) : R[X]_n :=
+  ⟨X ^ (i : ℕ), X_pow_mem_degreeLT i⟩
+
+@[simp] lemma xPow_val : (xPow R n i : R[X]) = X ^ (i : ℕ) :=
+  rfl
+
+@[simp] lemma basis_repr : (basis R n).repr P i = (P : R[X]).coeff i :=
+  rfl
+
+@[simp] lemma basis_apply (i : Fin n) :
+    basis R n i = xPow R n i := by
+  rw [Basis.apply_eq_iff]; ext j
```

Comment:
```suggestion
  rw [Basis.apply_eq_iff]
  ext j
```

Reply:
done. what is the convention for this?

Reply:
I think `;` is discouraged in general if done without a reason.